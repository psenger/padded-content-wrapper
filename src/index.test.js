import ContentWrapper from '.'
import React, { useState } from 'react'
import renderer, { act } from 'react-test-renderer'

const ProxyContentWrapper = (props) => {
  const [expanded, setExpanded] = useState(false)
  return (
    <div onClick={() => setExpanded(!expanded)} id='test-harness-proxy'>
      <ContentWrapper {...props} expanded={expanded} id='test-harness' />
    </div>
  )
}

describe('ContentWrapper', () => {
  test('is truthy', () => {
    const component = renderer.create(
      <ProxyContentWrapper>hello world</ProxyContentWrapper>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    act(() => {
      /* fire events that update state */
      tree.props.onClick()
    })

    expect(component.toJSON()).toMatchSnapshot()
  })
})
