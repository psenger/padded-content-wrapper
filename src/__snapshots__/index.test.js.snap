// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentWrapper is truthy 1`] = `
<div
  id="test-harness-proxy"
  onClick={[Function]}
>
  <div
    className="content-wrapper-collapsed"
    id="test-harness"
  >
    hello world
  </div>
</div>
`;

exports[`ContentWrapper is truthy 2`] = `
<div
  id="test-harness-proxy"
  onClick={[Function]}
>
  <div
    className="content-wrapper-expanded"
    id="test-harness"
  >
    hello world
  </div>
</div>
`;
